crates/subspace-core-primitives/src/lib.rs

```diff
@@ -263,6 +263,13 @@ impl FlatPieces {
     }
 }
 
+impl Extend<Piece> for FlatPieces {
+    fn extend<T: IntoIterator<Item = Piece>>(&mut self, iter: T) {
+        self.0
+            .extend(iter.into_iter().flat_map(|piece| piece.0.into_iter()))
+    }
+}
```

Comment:
For `unzip`

Reply:
Hm... it isn't efficient though, depending on number of elements this could allocate a lot. The correct solution would be to return `FlatPieces` from `get_sequential_pieces`, so I'd add a comment to remove this once we have those changes.

Reply:
We still would need to filter those, but it for sure would be more efficient

Reply:
I do not think we would need that. We'd read `piece_index_hash`es first, allocate big enough `FlatPieces` for the actual pieces once we know the size and then just read into it with `as_pieces_mut()` for instance.